### ISSUE TITLE
Allow box_dots to traverse BoxLists

### ISSUE THREAD

**Issue by NEOOOOOOOOOO** [cdgriffith/Box#128]
Windows 10, Python 3.6.6, Box 4.0.4

```
>>> import box
>>> my_box = Box({'b': {'c': [{'d': 'my_value'}, {'e': 'my_value'}]}}, box_dots=True)
>>> print(my_box.b.c[1].e)
my_value
>>> print(my_box['b.c[1].e']) 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Python36\lib\site-packages\box\box.py", line 330, in __getitem__
    return self[first_item][children]
  File "C:\Python36\lib\site-packages\box\box.py", line 334, in __getitem__
    raise BoxKeyError(str(err)) from None
box.exceptions.BoxKeyError: "'c[1].e'"
```

Can I get the value by using dot notaion when key in a list?

Thanks!

**Comment by cdgriffith** [comment]
Adding in commit https://github.com/cdgriffith/Box/commit/5241fe2880899e8b3b5eed7a1a1f5998ca10c59d aiming for 4.1.0 release

**Comment by cdgriffith** [comment]
slated for 4.1 https://github.com/cdgriffith/Box/pull/132

**Comment by cdgriffith** [comment]
Addressed in [4.1.0](https://github.com/cdgriffith/Box/tree/4.1.0)
